### PR TITLE
DW-505 remove gulp dist task

### PR DESCRIPTION
**Pull request by cbernat**
### Remove gulp from ui library
Replace gulp dist by npm dist task

Later in another PR the replace would be extensive to build task. I've moved on with dist because it's simpler.

https://cdn.fromdoppler.com/doppler-ui-library/build1031/index.html

#Checklist

- [x] Check if there is a new color.
- [x] Create the color variable in the _colors.scss file.
- [x] Follow the color nomenclature.
- [x] Create the color class with its variable.
- [x] Create the color component in the html, show the name, color class and its exadecimal.

### CODE REVIEW

File: assets/scss/core/_settings.scss [hunk 1]
```diff
@@ -4,9 +4,6 @@
 /// @group Settings
 ////
 
-///
-@import '../helpers/colors';
```
Comment (cbernat) [inline assets/scss/core/_settings.scss:8]: Colors were import more than once, check in styles.scss that I've moved it up.

File: assets/scss/modules/_tooltip.scss [hunk 2]
```diff
@@ -6,21 +6,17 @@
 /// Classes for the tooltip component
 
 .dp-library {
-  .icon-tip-icon {
```
Comment (cbernat) [inline assets/scss/modules/_tooltip.scss:9]: This I believe is from another PR right @GusBaamonde ?

Comment (GusBaamonde) [reply assets/scss/modules/_tooltip.scss:9]: :open_mouth:

**Comment by cbernat** [comment]
@andresmoschini this is nearly finished what do you think? pretty messy O_o

**Comment by andresmoschini** [comment]
> @andresmoschini this is nearly finished what do you think? pretty messy O_o

I did not review it deeply, but, at a glance, it does not look messy. In my opinion, it seems cleaner than the gulp alternative.

File: gulpfile.js [hunk 3]
```diff
@@ -385,10 +385,6 @@ gulp.task('run', ['clean', 'serve'], function() {
   util.log('Done!');
 });
 
-// Watch for changes
-gulp.task('watch', ['build'], function() {
-  gulp.watch(config.folderAssets.base + '/**/*.scss', ['sass']);
-});
```
Comment (andresmoschini) [inline gulpfile.js:391]: You should remove the hole file, right?

Comment (cbernat) [reply gulpfile.js:391]: Not right now, only dist task is obsolete in this file, I'm doing this step by step, else it's a mess.

Comment (andresmoschini) [reply gulpfile.js:391]: What a deception! :P

File: assets/process-html.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+var { exec } = require('child_process');
+var path = require("path");
+
+var glob = require("glob");
+var env = process.argv.slice(2)[0];
+
+
```
Comment (matiasbeckerle) [inline assets/process-html.js:7]: Quite minor but there's an unnecessary empty line.

File: assets/process-html.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+var { exec } = require('child_process');
+var path = require("path");
+
+var glob = require("glob");
```
Comment (matiasbeckerle) [inline assets/process-html.js:4]: Super minor but it seems you use a mixture of `"` and `'` I think you should pick one and stick with it.

File: assets/process-html.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+var env = process.argv.slice(2)[0];
+
+
+glob("./assets/templates/*.html", '', function (er, files) {
+    files.forEach(function(file){
+    exec('npm run html-processor ' + file + ' -- --output ./'+env+'/'+path.basename(file)+' -e '+env, (err, stdout, stderr) => {
```
Comment (matiasbeckerle) [inline assets/process-html.js:10]: Is there any chance of using template literals here?

File: assets/process-html.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        if (err) {
+          //some err occurred
```
Comment (matiasbeckerle) [inline assets/process-html.js:12]: I think this one is unnecessary.

File: package.json [hunk 8]
```diff
@@ -23,11 +24,51 @@
...
+    "eclint-check": "./node_modules/.bin/eclint check \"**/*\"",
+    "font": "make-dir ./assets/scss/libs/webfont/compiled && webfont ./assets/icons/*.svg -d ./assets/fonts -s ./assets/scss/libs/webfont/compiled --font-height 1001 --normalize true",
+    "font-dist": "npm run font && npm run copyfiles ./assets/fonts/*.* ./dist/fonts",
+    "html-dist": "node ./assets/process-html.js dist",
+    "html-processor": "./node_modules/.bin/htmlprocessor",
+    "imagemin": "./node_modules/.bin/imagemin",
+    "imagemin-dist": "imagemin ./assets/img/* --out-dir ./dist/img",
+    "js-copy-push-dist": "npm run copyfiles ./assets/firebasePush.js ./assets/firebase-messaging-sw.js ./dist",
+    "js-minify": "./node_modules/.bin/uglifyjs ./assets/js/*.js --compress --mangle",
+    "js-minify-dist": "make-dir ./dist/js && npm run --silent js-minify > ./dist/js/app.js",
```
Comment (matiasbeckerle) [inline package.json:65]: Just asking from the ignorance point of view. Isn't there any fancy way of doing all these tasks without having to declare a regular one vs the dist(ributable) one? I'm quite rusty but maybe this has been improved in the recent years?

**Review by matiasbeckerle — APPROVED** [review]
LGTM nice work!

File: assets/process-html.js (hunk 6 above)
Comment (cbernat) [reply assets/process-html.js:10]: Not sure but I'll try!

File: package.json (hunk 8 above)
Comment (cbernat) [reply package.json:65]: I'll check, @matiasbeckerle thanks a lot for you feedback!

Comment (cbernat) [reply package.json]: Done!

**Comment by cbernat** [comment]
test this please.

**Comment by andresmoschini** [comment]
Thanks for your review @matiasbeckerle! we love you!

File: package.json (hunk 8 above)
Comment (matiasbeckerle) [reply package.json]: Niiice, thanks for investigating this!

**Pull request merged**